src/lib.rs

```diff
@@ -241,6 +241,26 @@ pub trait PrimeField: Field + From<u64> {
         Some(res)
     }
 
+    /// Obtains a field element congruent to the integer `v`.
+    ///
+    /// For fields where `Self::CAPACITY >= 128`, this is injective and will produce a
+    /// unique field element.
+    ///
+    /// For fields where `Self::CAPACITY < 128`, this is surjective; some field elements
+    /// will be produced by multiple values of `v`.
+    ///
+    /// If you want to deterministically sample a field element representing a value, use
+    /// [`FromUniformBytes`] instead.
+    fn from_u128(v: u128) -> Self {
+        let lower = v as u64;
+        let upper = (v >> 64) as u64;
+        let mut tmp = Self::from(upper);
+        for _ in 0..64 {
+            tmp = tmp.double();
+        }
```

Comment:
Ideally I would like Field to support shifts (`x << w` $= x \cdot 2^w$ and `x >> w` $= x \cdot 2^{-w}$).

Reply:
This was issue #60, and no one disagreed with my counter-argument, so I closed the issue. If you have a motivating use case or argument, please share it there.

Reply:
I hadn't noticed that had been closed; I'll reply there.